ts/parts/Chart.ts

```diff
@@ -697,13 +697,29 @@ extend(Chart.prototype, /** @lends Highcharts.Chart.prototype */ {
         plotY: number,
         inverted?: boolean
     ): boolean {
-        var x = inverted ? plotY : plotX,
-            y = inverted ? plotX : plotY;
+        var chart = this,
+            x = inverted ? plotY : plotX,
+            y = inverted ? plotX : plotY,
+            center,
+            isInsidePlot;
+
+        if (chart.polar) {
```

Comment:
The core Highcharts functionality shouldn't deal with polar chart functionality. Instead, all this functionality should be gathered in the Polar.ts file. You can resolve that by implementing an event pattern like seen with the `getAxes` and `afterDrawChartBox` events in Polar.ts. In order to modify a return value, the return value must be part of event arguments, something like this:

```js
    isInsidePlot: function (
        this: Highcharts.Chart,
        plotX: number,
        plotY: number,
        inverted?: boolean
    ): boolean {
        var x = inverted ? plotY : plotX,
            y = inverted ? plotX : plotY;

        var e = {
            isInsidePlot: x >= 0 &&
                x <= this.plotWidth &&
                y >= 0 &&
                y <= this.plotHeight
        };
        // A hook for an event listener can mutate e.isInsidePlot
        fireEvent(this, 'isInsidePlot', e);

        return e.isInsidePlot;
    },
```